### PR TITLE
test(release): harden 0.13.3 release reliability

### PR DESCRIPTION
**Pull request by Shishir435**
## Summary

- bump the extension and all workspace packages to `0.13.3`
- add a release-integrity regression test so workspace versions cannot drift and `CHANGELOG.md` cannot silently become binary/NUL-corrupted again
- promote fragmented/malformed provider streaming and HTTP/connection failure coverage from the scheduled provider matrix into the blocking Chromium critical gate
- preserve the existing fully configurable custom-provider URL coverage; no provider hostname restrictions are introduced

## Release lineage

`release/0.13.3` was created directly from the cleaned `release/0.13.2` head after #319 restored the 0.13.2 changelog text. This PR targets `release/0.13.3`; it does not add new work to the 0.13.2 release candidate.

## Intent

Keep 0.13.3 as a small maintenance/reliability patch rather than starting another architecture or feature cycle. The new blocking browser cases cover two failure classes that unit/provider-contract tests cannot fully prove: real fetch/SSE chunk boundaries and durable failure propagation through the packaged extension runtime.

## Validation

CI should run the repository's static, unit/coverage, package, bundle, Chromium/Firefox, migration, and distribution gates. The newly promoted provider cases are now part of `@critical` rather than weekly/manual-only coverage.

<h3>Greptile Summary</h3>

The PR hardens the 0.13.3 release process and promotes provider failure-path coverage into the blocking Chromium suite.
- Bumps the root extension and workspace packages to 0.13.3.
- Adds current-release changelog and text-integrity checks.
- Promotes fragmented SSE and provider failure scenarios to the critical browser gate.
- Enables an initial Biome cognitive-complexity warning threshold.

<h3>Confidence Score: 5/5</h3>

The PR appears safe to merge.

No blocking failure remains.

<details><summary><h3>Important Files Changed</h3></summary>

| Filename | Overview |
|----------|----------|
| src/lib/__tests__/release-integrity.test.ts | The revised positional heading check closes the previously reported stale-version loophole without duplicating workspace discovery. |
| CHANGELOG.md | Adds the 0.13.3 release entry and restores contiguous comparison links for recent releases. |
| e2e/chromium/critical/__tests__/provider-streaming.spec.ts | Promotes existing streaming and connection-failure scenarios into the blocking critical test selection. |
| biome.json | Adds a warning-level cognitive-complexity ceiling without changing build-blocking behavior. |
| package.json | Updates the extension version to 0.13.3 consistently with all changed workspace manifests. |

</details>

<sub>Reviews (4): Last reviewed commit: ["chore(lint): baseline cognitive complexi..."](https://github.com/shishir435/ollama-client/commit/e9734398b3eb9aceddef3693e2fbc23be8a7c253) | [Re-trigger Greptile](https://app.greptile.com/api/retrigger?id=57515293)</sub>

### CODE REVIEW

File: src/lib/__tests__/release-integrity.test.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+      )
+    }
+  })
+
+  it("keeps the changelog textual and structurally usable", () => {
+    const changelog = readText("CHANGELOG.md")
+
+    expect(changelog).not.toContain("\u0000")
+    expect(changelog).toContain("# Changelog")
+    expect(changelog).toContain("## [Unreleased]")
```
[the remark this answers is not in the document]
Comment (Shishir435) [reply src/lib/__tests__/release-integrity.test.ts]: Fixed. The integrity test now requires the current package version to be the first release heading immediately after `Unreleased`, and `CHANGELOG.md` now has a real 0.13.3 entry. This prevents an older heading such as 0.13.2 from satisfying the check.

File: src/lib/__tests__/release-integrity.test.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+import { describe, expect, it } from "vitest"
+
+const ROOT = path.resolve(import.meta.dirname, "../../..")
+const PACKAGE_FILES = [
+  "package.json",
+  "packages/contracts/package.json",
+  "packages/runtime-core/package.json",
+  "packages/chat-runtime/package.json",
+  "packages/olc/package.json"
+] as const
```
[the remark this answers is not in the document]
Comment (Shishir435) [reply src/lib/__tests__/release-integrity.test.ts]: Fixed by removing the duplicate hard-coded workspace package list from the new test. Workspace version alignment remains owned by the existing dynamic `config/__tests__/package-versions.test.ts`; the new test now covers only changelog/text integrity.
